lib/dry/system/errors.rb

```diff
@@ -92,5 +92,36 @@ def initialize(plugin, message, gem = nil)
         super("dry-system plugin #{plugin.inspect} failed to load its dependencies: #{details}")
       end
     end
+
+    # Exception raised when auto-registerable component is not loadable
+    #
+    # @api public
+    ComponentNotLoadableError = Class.new(NameError) do
+      # @api private
+      def initialize(component, error, corrections: DidYouMean::ClassNameChecker.new(error).corrections)
+        full_class_name = [error.receiver, error.name].join("::")
+
+        message = [
+          "Component '#{component.key}' with auto_register: true is not loadable.",
+          "Looking for #{full_class_name}."
+        ]
+
+        if corrections.any?
+          case_correction = corrections.find { |correction| correction.casecmp?(full_class_name) }
+          if case_correction
+            acronyms_needed = case_correction.split("::").difference(full_class_name.split("::"))
```

Comment:
`Array#difference` isn't very popular method, but... it's what we need! 

By doing it this direction (instead of the opposite), we get the correct existing (acronym-ized) part of the namespace. If we do it the other direction, we'd need to `.upcase` the result.